addons/udes_security/tests/test_desktop_access.py

```diff
@@ -104,3 +98,140 @@ def test_all_users_can_access_database(self):
+            url, timeout=timeout, headers=headers, allow_redirects=allow_redirects
+        )
+
+    def load_action(self, action_id):
+        """
+        Helper function to simulate loading an action via URL
+
+        :param: action_id: int() for an action (ir.actions.server, ir.actions.act_window, etc..)
+
+        :return: requests.models.Response() object
+        """
+        return self.url_open_json(
+            url=f"{self.base_url}/web/action/load",
+            data={
+                "params": {
+                    "Content-Type": "application/json",
+                    "action_id": action_id,
+                    "additional_context": {},
+                },
+                "csrf_token": http.WebRequest.csrf_token(self),
+            },
+        )
+
+    def test_access_to_actions(self):
+        """
+        Load various actions as both user_1 and user_2
+        and ensure errors are raised accordingly
+        """
+        # Settings/Technical/Sequences & Identifiers/Sequences (ir.actions.act_window)
+        sequence_action = self.env.ref("base.ir_sequence_form")
+        # Settings/Users & Companies/Users (ir.actions.act_window)
+        users_action = self.env.ref("base.action_res_users")
+        # Discuss (ir.actions.client)
+        discuss_action = self.env.ref("mail.action_discuss")
+        # Apps/Apps/Third-Party Apps (ir.actions.act_url)
+        third_party_apps_action = self.env.ref("base.menu_third_party")
+        # DEBT: There are no ir.actions.server, or ir.actions.report actions available,
+        # in future add them to setUpClass
+        user_action_access_list = [
+            {
+                "user": self.user_1,
+                "action": sequence_action,
+                "has_access": True,
+            },
+            {
+                "user": self.user_2,
+                "action": sequence_action,
+                "has_access": False,
+            },
+            {
+                "user": self.user_1,
+                "action": users_action,
+                "has_access": True,
+            },
+            {
+                "user": self.user_2,
+                "action": users_action,
+                "has_access": False,
+            },
+            {
+                "user": self.user_1,
+                "action": discuss_action,
+                "has_access": True,
+            },
+            {
+                "user": self.user_2,
+                "action": discuss_action,
+                "has_access": True,
+            },
+            {
+                "user": self.user_1,
+                "action": third_party_apps_action,
+                "has_access": False,
+            },
+            {
+                "user": self.user_2,
+                "action": third_party_apps_action,
+                "has_access": False,
+            },
+        ]
+        access_error_text = "odoo.exceptions.AccessError"
+        for user_action_access_dict in user_action_access_list:
```

Comment:
Nit: use subTest? It would be nice but I won't block the review if you don't implement it.

Reply:
sorry not sure what you're asking - I'm using subTest a few lines later

Reply:
Oh, how did I not see it? 😆

Reply:
updated it to have a msg either way which is helpful if failures happen